pkg/scheduler/priorities.go

```diff
@@ -132,3 +134,92 @@ func (n *NodeLabelPrioritizer) CalculateNodeLabelPriority(pod api.Pod, podLister
 	}
 	return result, nil
 }
+
+// BalancedResourceAllocation is favors nodes with balanced resource usage rate.
```

Comment:
Did you get this algorithm from this paper? "An Energy Efficient Virtual Machine Placement
Algorithm with Balanced Resource Utilization" http://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=6603690

Please document if so-- it explains how to extend the math to further dimensions.